models/catalog.cattle.io.app.js

```diff
@@ -215,13 +215,7 @@ export default class CatalogApp extends SteveModel {
       const mcapps = await this.$dispatch('management/findAll', { type: MANAGEMENT.MULTI_CLUSTER_APP }, { root: true });
 
       if (mcapps) {
-        for (let i = 0 ; i < mcapps.length ; i++) {
-          const res = mcapps[i].spec?.targets?.find(target => target.appName === this.metadata?.name);
-
-          if (res) {
-            return res;
-          }
-        }
+        return mcapps.find(mcapp => mcapp.spec.targets.find(target => target.appName === this.metadata?.name));
```

Comment:
We've lost the ? on spec.targets - I'd put that back in, just in case.

Reply:
I've reapplied the `spec?.targets?` that was there before.